### PR TITLE
Check expiration with TTL of column family

### PR DESCRIPTION
**Pull request by ummae**
#1445 

DBWithTTL does not return expired data anymore even if before compaction run.

There are two minor points(maybe big)

1) In `MultiGet`, we got multiple column families, but it check expiration of whole values with the ttl of first column family's. I don't have simple but elegant idea to correct this (without modify base implementation of MultiGet)

2) `TtlIterator` still return staled data.

any advice?

### CODE REVIEW

File: utilities/ttl/db_ttl_impl.cc [hunk 1]
```diff
@@ -216,6 +240,11 @@ std::vector<Status> DBWithTTLImpl::MultiGet(
     if (!statuses[i].ok()) {
       continue;
     }
+
+    if (IsStale((*values)[0], column_family_name)) {
```
Comment (igorcanadi) [inline utilities/ttl/db_ttl_impl.cc]: hmm, why don't you do the check for each value here?

**Comment by igorcanadi** [comment]
Hmm, I actually have a bigger concern. Now you added a hash table lookup on every single Get(), which could regress performance, especially with in-memory RocksDB. At the very least we should add an option to enable user to chose this new behavior.

@siying, @IslamAbdelRahman  what do you think about this patch?

**Comment by ummae** [comment]
I was worried about performance loss too, but I found [DateTieredDB](https://github.com/facebook/rocksdb/blob/master/utilities/date_tiered/date_tiered_db_impl.cc#L256) lookup hash table for Get operation too(the key type is int, though).

**Comment by igorcanadi** [comment]
> I was worried about performance loss too, but I found DateTieredDB lookup hash table for Get operation too(the key type is int, though).

Ha, interesting. :) Can we add a fast-path for default column family? That way people with a single column family (most of them) are not affected.

File: utilities/ttl/db_ttl_impl.cc (hunk 1 above)
Comment (ummae) [inline utilities/ttl/db_ttl_impl.cc]: it's typo..

**Comment by ummae** [comment]
I have an idea, maybe we can use int [GetID](https://github.com/facebook/rocksdb/blob/master/include/rocksdb/db.h#L76) of `ColumnFamilyHandle` instead its name if we can guarantee that uniqueness of IDs in single a db. What do you think?

**Comment by ummae** [comment]
[GetID](https://github.com/facebook/rocksdb/blob/master/db/column_family.cc#L1017) does not seem to be duplicated.

**Comment by siying** [comment]
Can you make this an option of TTL DB and it is by default off? Now the behavior fits caching use cases where some more results don't hurt.

**Comment by ummae** [comment]
@siying No problem. By the way, could you let me know any hints why travis-ci tests failed? it seems not related with the change list.

**Comment by gfosco** [comment]
Closing this via automation due to lack of activity.  If discussion is still needed here, please re-open or create a new/updated issue.

**Pull request closed without merging**